static/app/components/tooltip.tsx

```diff
@@ -225,7 +236,18 @@ class Tooltip extends React.Component<Props, State> {
 
     propList.containerDisplayMode = this.props.containerDisplayMode;
     return (
-      <Container {...propList} className={this.props.className} ref={ref}>
+      <Container
+        {...propList}
+        className={this.props.className}
+        ref={el => {
+          if (typeof ref === 'function') {
+            ref(el);
+          }
+          if (showOnEllipsis) {
+            this.isEllipsis = Boolean(el && isOverflown(el));
```

Comment:
How does this update itself when the element is resized?

Reply:
That's a good question that I'm also trying to understand for myself. It seems like hovering over the text triggers a re-render (toggles `isOpen` in the state) and in turn re-renders the triggers/content

Reply:
https://user-images.githubusercontent.com/22846452/151446495-1556ddbb-075d-41c5-8589-58cf8f1e1587.mov

Checking it out with the profiler says that's why `Tooltip` rerendered. Then I think it sets `tip` to be the `Popper` component and that's why it's saying `children` is the reason `Manager` re-rendered